docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+|----------------------	|----------------------------------------------------------------------------------------------------------------------------------------------------------------------	|----------------------------	|
+| PORT                 	| The port on which the middleware will listen <br>for job updates or results from the CL node                                                                         	| 3000                       	|
+| EI_CHAINLINKURL      	| The CL node URL in order to connect to its API<br>to listen for jobs and send job requests.<br><b>Note that this has no default value and needs<br>to be defined</b> 	| N/A                        	|
+| FROM                 	| The address of the oracle from which to push prices.<br><b>Note that this has no default value and needs<br>to be defined</b>                                        	| N/A                        	|
+| SUBMIT_RETRIES       	| The number of retries to try when submitting a price<br>on-chain and it fails                                                                                        	| 3                          	|
+| BLOCK_INTERVAL       	| The block time of the chain in seconds. This is used<br>to query the price and round at every interval.                                                              	| 6                          	|
+| SEND_CHECK_INTERVAL  	| The interval in seconds which is waited between each send.                                                                                                           	| 12                         	|
+| AGORIC_RPC           	| The Agoric's node RPC endpoint                                                                                                                                       	| http://0.0.0.0:26657       	|
+| STATE_FILE           	| The path to the middleware state's file                                                                                                                              	| data/middleware_state.json 	|
+| CREDENTIALS_FILE     	| The path to the file containing the credentials to the <br>CL node                                                                                                   	| config/ei_credentials.json 	|
+
+
+The CREDENTIALS_FILE should contain a JSON object containing the credentials to communciate wioth the CL node as can be seen below
+```json
+{
+  "EI_IC_ACCESSKEY": "",
+  "EI_IC_SECRET": ""
+}
+```
+
+
+<br>
+<div id='readState'></div>
+
+<b>readState()</b>
+
+Use: This function is used to read the middleware state from the file
+
+Returns: The middleware's state as JSON from the file or an empty initialised state
+
+What it does:
+  1. Tries to read the file using <b>readJSONFile</b>
+  2. If it succeeds, it returs the state
```

Comment:
```suggestion
  2. If it succeeds, it returns the state
```

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/e594b39b5953181d0cb19f39b70c1bd9a48202bc